### PR TITLE
Update x2goclient to 4.1.2.1

### PR DESCRIPTION
**Pull request by ytbmulder**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/x2goclient.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
 cask 'x2goclient' do
-  version '4.1.1.0'
+  version '4.1.2.1'
```
Comment (adidalal) [inline Casks/x2goclient.rb]: Please change it to `version '4.1.2.1, 20180717'` and use version methods, as version seems to be linked to date implicitly here.

Info [here][https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/version.md]

**Review by adidalal — CHANGES_REQUESTED** [review]

File: Casks/x2goclient.rb (hunk 1 above)
Comment (ytbmulder) [reply Casks/x2goclient.rb]: I fixed the version as requested, added a cask for Snow Leopard, and added a caveats section.

File: Casks/x2goclient.rb [hunk 2]
```diff
@@ -20,4 +25,16 @@
   app 'x2goclient.app'
 
   zap trash: '~/Library/Preferences/x2goclient.plist'
+
+  caveats "The X2Go OS X client uses the OS X X11 server.
```
Comment (vitorgalvao) [inline Casks/x2goclient.rb]: `caveats` are for instructions related to installation, not usage. Please remove it.

[This is documented](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats).

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/x2goclient.rb (hunk 2 above)
Comment (ytbmulder) [reply Casks/x2goclient.rb]: I removed the caveats section. Is there a place to inform the user about the cask's usage?

Comment (vitorgalvao) [reply Casks/x2goclient.rb]: > Is there a place to inform the user about the cask's usage?

The software’s usage, not the cask’s usage, and therein lies the difference. As per the linked documentation, `caveats` are for installation-related matters (i.e. cask’s usage). For the software’s usage, the user has to consult the vendor page. [This is also documented](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/not_a_discoverability_service.md).

Ultimately we manage packages: we make it easier for you to install and uninstall things. It’s your responsibility to know how to use them.

Comment (ytbmulder) [reply Casks/x2goclient.rb]: Thank you for the extensive reply. This makes sense now.

Comment (ytbmulder) [reply Casks/x2goclient.rb]: @vitorgalvao I have made the requested changes.

**Pull request merged**